corehq/apps/app_manager/xform.py

```diff
@@ -970,6 +981,93 @@ def already_has_meta(self):
 
         return meta_blocks
 
+    def _add_usercase_bind(self, usercase_path):
+        self.add_bind(
+            nodeset=usercase_path + 'case/@case_id',
+            calculate=SESSION_USERCASE_ID,
+        )
+
+    def add_usercase(self, form):
+        from corehq.apps.app_manager.util import split_path
+
+        usercase_path = 'usercase/'
+        actions = form.active_actions()
+        usercase_bound = False
+
+        if 'update_case' in actions and hasattr(actions['update_case'], 'update'):
+            usercase_updates = get_usercase_keys(actions['update_case'].update.items())
+            if usercase_updates:
+                if not usercase_bound:
+                    self._add_usercase_bind(usercase_path)
+                    usercase_bound = True
+                usercase_block = _make_elem('{x}usercase')
+                case_block = CaseBlock(self, usercase_path)
+                case_block.add_update_block(usercase_updates)
+                usercase_block.append(case_block.elem)
+                self.data_node.append(usercase_block)
+
+        if 'case_preload' in actions:
+            self.add_casedb()
+            usercase_preloads = get_usercase_values(actions['case_preload'].preload.items())
+            if usercase_preloads and not usercase_bound:
+                self._add_usercase_bind(usercase_path)
+                usercase_bound = True
+            for nodeset, property_ in usercase_preloads.items():
+                parent_path, property_ = split_path(property_)
+                property_xpath = {
+                    'name': 'case_name',
+                    'owner_id': '@owner_id'
+                }.get(property_, property_)
+
+                id_xpath = get_case_parent_id_xpath(parent_path, case_id_xpath=SESSION_USERCASE_ID)
+                self.add_setvalue(
+                    ref=nodeset,
+                    value=id_xpath.case().property(property_xpath),
+                )
+
+    def add_usercase_advanced(self, form):
```

Comment:
I'm thinking we should handle the advanced forms differently and rather than mixing in the user case properties in other actions we should just have a separate action with only user case properties. That way we wouldn't have to add any new code to suite_xml.py or xform.py to for advanced forms / modules.

Can chat more about this offline.